src/editor/VideoViewer.js

```diff
@@ -0,0 +1,243 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var DocumentManager     = require("document/DocumentManager"),
+        VideoViewTemplate   = require("text!htmlContent/video-view.html"),
+        ProjectManager      = require("project/ProjectManager"),
+        PreferencesManager  = require("preferences/PreferencesManager"),
+        LanguageManager     = require("language/LanguageManager"),
+        MainViewFactory     = require("view/MainViewFactory"),
+        Strings             = require("strings"),
+        StringUtils         = require("utils/StringUtils"),
+        FileSystem          = require("filesystem/FileSystem"),
+        BlobUtils           = require("filesystem/impls/filer/BlobUtils"),
+        FileUtils           = require("file/FileUtils"),
+        _                   = require("thirdparty/lodash"),
+        Mustache            = require("thirdparty/mustache/mustache"),
+        Image               = require("editor/Image");
+
+
+    var _viewers = {};
+
+    // Get a Blob URL out of the cache
+    function _getVideoUrl(file) {
+        return BlobUtils.getUrl(file.fullPath);
+    }
+
+    /**
+     * Whether or not this is an image, or an SVG image (vs SVG XML file).
+     */
+    function isVideo(fullPath) {
+        var lang = LanguageManager.getLanguageForPath(fullPath);
+        var id = lang.getId();
+        return id === "video";
+    }
+
+    /**
+     * ImageView objects are constructed when an image is opened
+     * @see {@link Pane} for more information about where ImageViews are rendered
+     *
+     * @constructor
+     * @param {!File} file - The image file object to render
+     * @param {!jQuery} container - The container to render the image view in
+     */
+    function VideoView(file, $container) {
+        this.file = file;
+
+        this.$el = $(Mustache.render(VideoViewTemplate, {
+            videoUrl: _getVideoUrl(file),
+            Strings: Strings
+        }));
+
+        console.log("Before timeout");
+        console.log(_getVideoUrl(file));
+        var that = this;
+
+        setTimeout(function(){
+            console.log("After timeout");
+            console.log(_getVideoUrl(file));
+            that.$videoEl.attr("src", _getVideoUrl(that.file));
+        }, 1000);
+
+        $container.append(this.$el);
+
+        this._naturalWidth = 0;
+        this._naturalHeight = 0;
+
+        this.relPath = ProjectManager.makeProjectRelativeIfPossible(this.file.fullPath);
+
+        // this.$imagePath = this.$el.find(".image-path");
+        this.$videoEl = this.$el.find("video");
+        this.$videoData = this.$el.find(".video-data");
+
+        this.$videoEl.on("canplay", _.bind(this._onVideoLoaded, this));
+        this.$videoEl.on("error", _.bind(console.error, console));
```

Comment:
I wonder if we should do more here.  Specifically, what if the current browser can't play the given media type?  We probably should do some kind of UI when things break at this level.  Video on the web is hard for people, since the encoders aren't really well understood.